### PR TITLE
Index refactoring

### PR DESCRIPTION
**Pull request by TrySound**
Everything is the same, but shorter :)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -29,75 +28,35 @@ module.exports = function(moduleName, options) {
...
+    var currentSearchPath = joinPath(splitSearchPath);
+
+    return Promise.resolve(null).then(function() {
+      if (!options.packageProp) return;
+      return loadPackageProp(currentSearchPath, options.packageProp);
+    }).then(function(result) {
+      if (result || !options.rc) return result;
+      return loadRc(path.join(currentSearchPath, options.rc), {
+        strictJson: options.rcStrictJson,
+      });
```
Comment (davidtheclark) [inline index.js]: Oooo, this is cool: taking advantage of the fact that you can return a synchronous value *or* a Promise and they are treated the same way. Working on this module has taught me a lot about Promises :)

**Comment by davidtheclark** [comment]
Nice job, thanks!

File: index.js (hunk 1 above)
Comment (TrySound) [inline index.js]: Yeah, they are very cool! :)

**Comment by davidtheclark** [comment]
@TrySound : How are you feeling about the API now?

**Comment by TrySound** [comment]
Seems much better

File: index.js (hunk 1 above)
Comment (jeddy3) [inline index.js]: > Working on this module has taught me a lot about Promises :)

Likewise, I've learnt a lot from just seeing the evolution of the code both here and in stylelint's `standalone.js`. So, thanks for that :)

Comment (TrySound) [inline index.js]: My pleasure :)

**Pull request merged**